src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -318,6 +319,39 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
     {
         emitSize = EA_SCALABLE;
         opt      = emitter::optGetSveInsOpt(emitTypeSize(intrin.baseType));
+
+        switch (intrin.id)
+        {
+            case NI_Sve_GetFfrByte:
+            case NI_Sve_GetFfrInt16:
+            case NI_Sve_GetFfrInt32:
+            case NI_Sve_GetFfrInt64:
+            case NI_Sve_GetFfrSByte:
+            case NI_Sve_GetFfrUInt16:
+            case NI_Sve_GetFfrUInt32:
+            case NI_Sve_GetFfrUInt64:
+            {
+                if ((intrin.op1 != nullptr) && ((intrin.op1->gtFlags & GTF_SPILLED) != 0))
+                {
+                    // If there was a op1 for this intrinsic, it means FFR is consumed here
+                    // and we need to unspill.
+                    unspilledFfr = true;
+                }
+                break;
+            }
+            case NI_Sve_LoadVectorFirstFaulting:
+            {
+                if ((intrin.op3 != nullptr) && ((intrin.op3->gtFlags & GTF_SPILLED) != 0))
+                {
+                    // If there was a op3 for this intrinsic, it means FFR is consumed here
+                    // and we need to unspill.
+                    unspilledFfr = true;
+                }
```

Comment:
So we don't call `genConsumeReg` on this operand? Is `GTF_SPILLED` the only thing that needs to be handled then?

I don't really understand what the FFR register is, but what happens if it gets changed by a callee and we didn't end up spilling the local (that represents p14) around that call. Is that possible?

Reply:
I guess we do call it, and this is detecting if it unspilled into `p14`? I wonder if unspilling itself could have the code to mirror p14 to ffr. I'm sort of worried about what can happen with GT_RELOAD or GT_COPY here... You may want to run jitstressregs (not sure if we'll have any coverage, though?)

Reply:
>but what happens if it gets changed by a callee and we didn't end up spilling the local (that represents p14) around that call. Is that possible?

That's the entire purpose of this PR, to make sure that we spill it around the call. Since all predicate registers are callee trash, using `P14` does force to save across the call (if it is live).

yes, unspilling will happen as part of `genConsumeMultiOpOperands()`. 

> I wonder if unspilling itself could have the code to mirror p14 to ffr

I could, but that will exercise in the common code path. Having it here already restricts us to do this for scalable APIs which are SVE only.

> I'm sort of worried about what can happen with GT_RELOAD or GT_COPY here... You may want to run jitstressregs (not sure if we'll have any coverage, though?)

Unfortunately we have basic coverage of the CI pipeline that I added few days back that tests with `AltJit` , but not with jitstressregs. I will see if I can test it locally.

Reply:
i didn't get a good repro, but I can potentially add a `gtSkipReloadOrCopy` here to bypass the COPY/RELOAD to check if i will be unspilling here or not. The `consumeRegs ` will anyway do the right thing.